diff --git a/lcls1_producers/smd_producer.py b/lcls1_producers/smd_producer.py
index 4c2fe87..dcdf7d4 100755
--- a/lcls1_producers/smd_producer.py
+++ b/lcls1_producers/smd_producer.py
@@ -654,7 +654,7 @@ for evt_num, evt in enumerate(ds.events()):
 
     #the ARP will pass run & exp via the enviroment, if I see that info, the post updates
     if ( (evt_num<100 and evt_num%10==0) or (evt_num<1000 and evt_num%100==0) or (evt_num%1000==0)):
-        if os.environ.get('ARP_JOB_ID', None) is not None:
+        if os.environ.get('ARP_JOB_ID', None) is not None and os.getenv("JID_UPDATE_COUNTERS") is not None:
             if ds.size == 1:
                 requests.post(os.environ["JID_UPDATE_COUNTERS"], json=[{"key": "<b>Current Event</b>", "value": evt_num+1}])
             elif ds.rank == 0:
@@ -703,7 +703,7 @@ if ds.rank==0:
 
 #finishing up here....
 logger.debug('rank {0} on {1} is finished'.format(ds.rank, hostname))
-if os.environ.get('ARP_JOB_ID', None) is not None:
+if os.environ.get('ARP_JOB_ID', None) is not None and os.getenv("JID_UPDATE_COUNTERS") is not None:
     if ds.size > 1:
         if ds.rank == 0:
             requests.post(os.environ["JID_UPDATE_COUNTERS"], json=[{"key": "<b>Last Event</b>", "value": "~ %d cores * %d evts"%(ds.size,evt_num)},{"key": "<b>Duration</b>", "value": "%f min"%(prod_time)}])
diff --git a/lcls2_producers/smd_producer.py b/lcls2_producers/smd_producer.py
index bb6c679..985132b 100755
--- a/lcls2_producers/smd_producer.py
+++ b/lcls2_producers/smd_producer.py
@@ -54,6 +54,7 @@ def define_dets(run, det_list):
     # Assumes that the config file with function parameters definition
     # has been imported under "config"
 
+    user_det_params = {}
     rois_args = []  # special implementation as a list to support multiple ROIs.
     dimgs_args = {}
     wfs_int_args = {}
@@ -68,6 +69,8 @@ def define_dets(run, det_list):
     pressio_compression_args = {}
 
     # Get the functions arguments from the production config
+    if "get_det_params" in dir(config):
+        user_det_params = config.get_det_params(run)
     if "getROIs" in dir(config):
         rois_args = config.getROIs(run)
     if "getDetImages" in dir(config):
@@ -100,6 +103,8 @@ def define_dets(run, det_list):
 
         # Common mode (default: None)
         common_mode = None
+        if detname in user_det_params and "cmpars" in user_det_params[detname]:
+            common_mode = user_det_params[detname]["cmpars"]
 
         if not havedet:
             continue
@@ -1057,7 +1062,7 @@ logger.debug("rank {0} on {1} is finished".format(rank, hostname))
 #        cmd = ['timestamp_sort_h5', h5_f_name, h5_f_name]
 
 if ds.unique_user_rank():
-    if os.environ.get("ARP_JOB_ID", None) is not None:
+    if os.environ.get("ARP_JOB_ID", None) is not None and os.getenv("JID_UPDATE_COUNTERS") is not None:
         requests.post(
             os.environ["JID_UPDATE_COUNTERS"],
             json=[
diff --git a/smalldata_tools/lcls2/DetObject.py b/smalldata_tools/lcls2/DetObject.py
index 6a8a79b..ae5bbbc 100644
--- a/smalldata_tools/lcls2/DetObject.py
+++ b/smalldata_tools/lcls2/DetObject.py
@@ -489,9 +489,17 @@ class Epix100Object(TiledCameraObject):
             30,
         ]  # Jacob (norm), Jacob, def, def(ami-like), mine, mine (norm), mine (norm-bank), none, raw, calib
         self.common_mode = kwargs.get("common_mode", self._common_mode_list[0])
-        if self.common_mode is None:
+        if isinstance(self.common_mode, tuple) or isinstance(self.common_mode, list):
+            self.common_mode = tuple(self.common_mode)
+            if len(self.common_mode) > 4:
+                print("Too many common mode parameters, taking only first 4!", self.common_mode)
+                self.common_mode = self.common_mode[:4]
+            elif len(self.common_mode) == 3:
+                print("Too few common mode parameters, but alg (cmpars[0]) is ignored. Extending!", self.common_mode)
+                self.common_mode = (0,) + self.common_mode
+        elif self.common_mode is None:
             self.common_mode = self._common_mode_list[0]
-        if self.common_mode not in self._common_mode_list:
+        elif self.common_mode not in self._common_mode_list:
             print(
                 "Common mode %d is not an option for as Epix detector, please choose from: "
                 % self.common_mode,
@@ -527,9 +535,13 @@ class Epix100Object(TiledCameraObject):
         mbits = 0  # do not apply mask (would set pixels to zero)
         # mbits=1 #set bad pixels to 0
 
+        if isinstance(self.common_mode, tuple):
+            self.evt.dat = self.det.raw.calib(
+                evt, cmpars=self.common_mode, rms=self.rms, normAll=True
+            )
         # Common Mode currently NOT working for ePix100
         # Skip all options that pass cmpars
-        if self.common_mode % 100 == 6:
+        elif self.common_mode % 100 == 6:
             # Was using cmpars=[6] -> This doesn't work for psana2
             # Some of these other kwargs may not do anything
             # self.evt.dat = self.det.raw.calib(
diff --git a/smalldata_tools/ana_funcs/azav_pyfai.py b/smalldata_tools/ana_funcs/azav_pyfai.py
index fafbced..6665003 100644
--- a/smalldata_tools/ana_funcs/azav_pyfai.py
+++ b/smalldata_tools/ana_funcs/azav_pyfai.py
@@ -5,9 +5,10 @@ try:
 except ModuleNotFoundError:
     print("pyFAI not available on LCLS-II env")
 
-from smalldata_tools.common.detector_base import DetObjectFunc
-from smalldata_tools.utilities import image_from_dxy
+from LCLSGeom.manager import get_geometry
+from LCLSGeom.converter import PsanaToPyFAI
 
+from smalldata_tools.common.detector_base import DetObjectFunc
 
 class azav_pyfai(DetObjectFunc):
     """
@@ -49,39 +50,16 @@ class azav_pyfai(DetObjectFunc):
     def __init__(self, **kwargs):
         self._name = kwargs.get("name", "pyfai")
         super(azav_pyfai, self).__init__(**kwargs)
-
         self.mask = kwargs.pop("userMask", None)
         self.threshold = kwargs.pop("thres", None)
         self.return2d = kwargs.pop("return2d", "False")
-
-        # azimuthal integrator argument
         self.poni_file = kwargs.pop("poni_file", None)
-        if self.poni_file is not None:
-            self.ai = pyFAI.load(str(self.poni_file))
-        else:
-            self._ai_kwargs = kwargs.pop("ai_kwargs", None)
-            assert (
-                self._ai_kwargs is not None
-            ), "Need either a calibration file or a set of keywords arguments to instantiate the pyFAI integrator."
-            # self.ai = pyFAI.azimuthalIntegrator.AzimuthalIntegrator(**self._ai_kwargs) # why does it not work?
-            self.ai = pyFAI.AzimuthalIntegrator(**self._ai_kwargs)
-        print(f"Created azimuthal integrator for {self._name}")
-        # we should have code that extract setup parameters
-        #    in case people mess w/ the input
-        # we should also extra q/theta/... bins for plotting later.
-        azcfg = self.ai.get_config()
-        dcfg = azcfg.pop("detector_config")
-        ocfg = dcfg.pop("orientation")
-        for k, v in azcfg.items():
-            setattr(self, "azcfg_%s" % k, v)
-        for k, v in dcfg.items():
-            setattr(self, "azcfg_det_%s" % k, v)
-        setattr(self, "azcfg_detorientation", ocfg.value)
-        setattr(self, "azcfg_qvals", self.ai.get_qa())
-        # setattr(self,'azcfg_qvals',self.ai.get_qa())
+        self._ai_kwargs = kwargs.pop("ai_kwargs", None)
+        if self._ai_kwargs is None and self.poni_file is None:
+            raise ValueError("Either a poni file or ai_kwargs must be provided to initialize the azimuthal integrator.")
 
         # integration arguments
-        self.pol_factor = kwargs.pop("polarization_factor", 1)
+        self.pol_factor = kwargs.pop("polarization_factor", -1)  # Vertical Polarization by default post Run 18
         self.npts = kwargs.pop("npts", 256)
         self.npts_az = kwargs.pop("npts_az", 360)
         # self.radial_range = kwargs.pop('radial_range',None)
@@ -94,27 +72,6 @@ class azav_pyfai(DetObjectFunc):
 
     def setFromDet(self, det):
         super(azav_pyfai, self).setFromDet(det)
-        # geometry
-        self.ix = det.ix
-        self.iy = det.iy
-
-        if self.poni_file is None:  # use calib value if calib file is given
-            if len(det.pixelsize) == 2:
-                if det.pixelsize[0] > 1:
-                    self.ai.pixel1 = det.pixelsize[0] * 1e-6
-                    self.ai.pixel2 = det.pixelsize[1] * 1e-6
-                else:
-                    self.ai.pixel1 = det.pixelsize[0]
-                    self.ai.pixel2 = det.pixelsize[1]
-            else:
-                if det.pixelsize[0] > 1:
-                    self.ai.pixel1 = det.pixelsize[0] * 1e-6
-                    self.ai.pixel2 = det.pixelsize[0] * 1e-6
-                else:
-                    self.ai.pixel1 = det.pixelsize[0]
-                    self.ai.pixel2 = det.pixelsize[0]
-
-        # use cmask by default. Note: pyFAI uses 0 for valid pixel.
         if (
             self.mask is not None
             and det.cmask is not None
@@ -135,12 +92,56 @@ class azav_pyfai(DetObjectFunc):
                     except:
                         self.mask = None
         if self.mask is not None and self.mask.ndim == 3:
-            maskInv = ~(self.mask.astype(bool))
-            self.mask = image_from_dxy(self.mask, self.ix, self.iy)
-            maskInvImg = image_from_dxy(maskInv, self.ix, self.iy)
-            maskImg = ~(maskInvImg)
-            self.mask = maskImg
-
+            self.mask = np.reshape(self.mask, (self.mask.shape[0]*self.mask.shape[1], self.mask.shape[2]))
+      
+        metrology = get_geometry(det.det.alias)
+        # instantiate PyFAI detector first 
+        # this adds the detector name mentioned in the poni file to pyFAI detector registry
+        # reading the poni file will not throw an error if the detector name is not in the registry
+        detector = PsanaToPyFAI.convert(metrology, det.det.alias)
+        # if detector is 3D (panels, slow, fast), then this scripts creates a fake 2D detector by
+        # squeezing the panel and slow dimensions.
+        # the 3D pixel layout is passed using the metrology data !
+        if self._ai_kwargs is None:
+            ai = pyFAI.load(self.poni_file)
+            self._ai_kwargs = ai.getPyFAI()
+
+        if self._units == "q_A^-1":
+            if "wavelength" not in self._ai_kwargs:
+                raise ValueError("Wavelength must be provided when using q_A^-1 units. Check poni file or ai_kwargs inputs.")
+            self.ai = pyFAI.AzimuthalIntegrator(
+                detector=detector,
+                dist=self._ai_kwargs["dist"],
+                poni1=self._ai_kwargs["poni1"],
+                poni2=self._ai_kwargs["poni2"],
+                rot1=self._ai_kwargs["rot1"],
+                rot2=self._ai_kwargs["rot2"],
+                rot3=self._ai_kwargs["rot3"],
+                wavelength=self._ai_kwargs["wavelength"],
+            )
+        else:
+            self.ai = pyFAI.AzimuthalIntegrator(
+                detector=detector,
+                dist=self._ai_kwargs["dist"],
+                poni1=self._ai_kwargs["poni1"],
+                poni2=self._ai_kwargs["poni2"],
+                rot1=self._ai_kwargs["rot1"],
+                rot2=self._ai_kwargs["rot2"],
+                rot3=self._ai_kwargs["rot3"],
+            )
+        # we should have code that extract setup parameters
+        #    in case people mess w/ the input
+        # we should also extra q/theta/... bins for plotting later.
+        azcfg = self.ai.get_config()
+        dcfg = azcfg.pop("detector_config")
+        ocfg = dcfg.pop("orientation")
+        for k, v in azcfg.items():
+            setattr(self, "azcfg_%s" % k, v)
+        for k, v in dcfg.items():
+            setattr(self, "azcfg_det_%s" % k, v)
+        setattr(self, "azcfg_detorientation", ocfg.value)
+        setattr(self, "azcfg_qvals", self.ai.get_qa())
+        # setattr(self,'azcfg_qvals',self.ai.get_qa())
         print(f"Azimuthal integrator:\n{self.ai}")
         return
 
@@ -149,9 +150,11 @@ class azav_pyfai(DetObjectFunc):
 
     def process(self, data):
         if self.threshold is not None:
-            data[data < threshold] = 0
+            data[data < self.threshold] = 0
         if data.ndim == 3:
-            data = image_from_dxy(data, self.ix, self.iy)
+            # instead of assembling the image, now we just squeeze the panel and slow dimensions
+            # so that image and detector have same shape.
+            data = np.reshape(data, (data.shape[0]*data.shape[1], data.shape[2]))
         out = self._process(data)
         return out
 
@@ -178,5 +181,4 @@ class azav_pyfai(DetObjectFunc):
                 method="cython",
                 **self._azav_kwargs,
             )
-
             return {"azav": I, "q": q}
