diff --git a/lcls1_producers/smd_producer.py b/lcls1_producers/smd_producer.py
index 4c2fe87..af212d8 100755
--- a/lcls1_producers/smd_producer.py
+++ b/lcls1_producers/smd_producer.py
@@ -654,7 +654,7 @@ for evt_num, evt in enumerate(ds.events()):
 
     #the ARP will pass run & exp via the enviroment, if I see that info, the post updates
     if ( (evt_num<100 and evt_num%10==0) or (evt_num<1000 and evt_num%100==0) or (evt_num%1000==0)):
-        if os.environ.get('ARP_JOB_ID', None) is not None:
+        if os.environ.get('ARP_JOB_ID', None) is not None and os.getenv("JID_UPDATE_COUNTERS") is not None:
             if ds.size == 1:
                 requests.post(os.environ["JID_UPDATE_COUNTERS"], json=[{"key": "<b>Current Event</b>", "value": evt_num+1}])
             elif ds.rank == 0:
diff --git a/lcls2_producers/smd_producer.py b/lcls2_producers/smd_producer.py
index 9f52b00..2b50c9b 100755
--- a/lcls2_producers/smd_producer.py
+++ b/lcls2_producers/smd_producer.py
@@ -1058,6 +1058,6 @@ logger.debug("rank {0} on {1} is finished".format(rank, hostname))
 #        cmd = ['timestamp_sort_h5', h5_f_name, h5_f_name]
 
 if ds.unique_user_rank():
-    if os.environ.get("ARP_JOB_ID", None) is not None:
+    if os.environ.get("ARP_JOB_ID", None) is not None and os.getenv("JID_UPDATE_COUNTERS") is not None:
         requests.post(
             os.environ["JID_UPDATE_COUNTERS"],
             json=[